examples/parallel_generator.py

```diff
@@ -0,0 +1,40 @@
+"""
+========================================
+Returning a generator in joblib.Parallel
+========================================
+
+This example illustrates memory optimization enabled by using
+ :class:`joblib.Parallel` to get a generator on the output of parallel jobs.
+ We first create tasks that are very memory demanding. We parallelize
```

Comment:
The gain is not so much when the task is memory consuming but when the returned object is large. I would rephrase this with `We first create a task that returns a result with a large memory footprint.`